File: c++/2565.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

vector<pair<int, int>> v;
int dp[100];

int main() {
  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  for (int i = 0; i < N; i++) {
    int a, b;
    cin >> a >> b;
    v.push_back({a, b});
  }

  sort(v.begin(), v.end());

  for (int i = 0; i < N; i++) {
    int k = 0;
    for (int j = 0; j < i; j++) {
      if (v[j].second < v[i].second) k = max(k, dp[j]);
    }
    dp[i] = k + 1;
  }

  int a = 0;
  for (int i = 0; i < N; i++) {
    a = max(a, dp[i]);
  }

  cout << N - a;

  return 0;
}

File: js/2565.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const N = +inputs[0];
let arr = [];
const dp = Array(N).fill(0);

for (let i = 1; i <= N; i++) {
  arr.push(inputs[i].split(' ').map(Number));
}

arr = arr.sort((a, b) => a[0] - b[0]).map((e) => e[1]);

for (let i = 0; i < N; i++) {
  let max = 0;
  for (let j = 0; j < i; j++) {
    if (arr[j] < arr[i]) max = Math.max(max, dp[j]);
  }
  dp[i] = max + 1;
}

const answer = N - Math.max(...dp);
console.log(answer);
